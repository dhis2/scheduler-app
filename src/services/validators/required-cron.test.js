import requiredCron from './required-cron'

describe('requiredCron', () => {
    it('should return an error message when there is no input', () => {
        const expected = 'A CRON expression is required'
        const actual = requiredCron([])

        expect(actual).toEqual(expect.stringMatching(expected))
    })

    it('should return an error message when input is not a string', () => {
        const expected = 'A CRON expression is required'
        const actual = requiredCron([])

        expect(actual).toEqual(expect.stringMatching(expected))
    })

    it('should return an error message when input is not a valid CRON expression', () => {
        const expected = 'Please enter a valid CRON expression'
        const actual = requiredCron('abc')

        expect(actual).toEqual(expect.stringMatching(expected))
    })

    it('should return undefined when input is a valid CRON expression', () => {
        const actual = requiredCron('* * * ? * *')

        expect(actual).toBeUndefined()
    })
})
