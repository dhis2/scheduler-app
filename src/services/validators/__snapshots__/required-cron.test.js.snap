// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requiredCron should return an error message when input is not a string 1`] = `"A CRON expression is required"`;

exports[`requiredCron should return an error message when input is not a valid CRON expression 1`] = `"Please enter a valid CRON expression"`;

exports[`requiredCron should return an error message when there is no input 1`] = `"A CRON expression is required"`;
